### PR TITLE
implement server provider reducer

### LINKED ISSUE #128: Implement server provider reducer

### PR DESCRIPTION
**Pull request by iLeonelPerea**
closes #128

### CODE REVIEW

File: src/components/provisionerForm/DigitalOceanService.js [hunk 1]
```diff
@@ -1,21 +1,45 @@
...
 import React, {PropTypes} from 'react';
 import {Link} from 'react-router';
 
-const DigitalOceanService = ( ) => {
+const DigitalOceanService = ( {serverProviderAppState, setServerProvider} ) => {
+  const handleDigitalOceanLogin = (e) => {
+    if(e.target.text != "Log out"){
+      setServerProvider({
+        name: 'digital_ocean',
+        token: '77e027c7447f468068a7d4fea41e7149a75a94088082c66fcf555de3977f69d3'
```
Comment (thetonymaster) [inline src/components/provisionerForm/DigitalOceanService.js]: what is this token?

File: src/test/serverProvider.spec.js [hunk 2]
```diff
@@ -6,11 +6,17 @@ import serverProvider from '../reducers/serverProvider';
...
+      name: 'digital_ocean',
+      token: '77e027c7447f468068a7d4fea41e7149a75a94088082c66fcf555de3977f69d3'
+    }};
     const nextState = serverProvider(initialState, action);
 
     expect(nextState).to.equal(fromJS({
-      server_provider: 'digital_ocean'
+      server_provider: {
+        name: 'digital_ocean',
+        token: '77e027c7447f468068a7d4fea41e7149a75a94088082c66fcf555de3977f69d3'
```
Comment (thetonymaster) [inline src/test/serverProvider.spec.js]: Same with this

File: src/components/provisionerForm/DigitalOceanService.js (hunk 1 above)
Comment (iLeonelPerea) [inline src/components/provisionerForm/DigitalOceanService.js]: isn't a real toke, just a string

File: src/test/serverProvider.spec.js (hunk 2 above)
Comment (iLeonelPerea) [inline src/test/serverProvider.spec.js]: same :D

File: src/components/provisionerForm/DigitalOceanService.js (hunk 1 above)
Comment (thetonymaster) [inline src/components/provisionerForm/DigitalOceanService.js]: I need you to walk me through this code to understand why you have this hardcoded variable.

Comment (thetonymaster) [inline src/components/provisionerForm/DigitalOceanService.js]: Replace it with the source of the variable.

**Comment by thetonymaster** [comment]
Looks temporary good to me.

**Pull request merged**